File: GetLeads.py
import httplib
import urlparse

#this is a sample call to base your API calls off of.
def main(url, headers):
  domain = urlparse.urlparse(url).netloc
  connection = httplib.HTTPSConnection(domain)
  connection.request("GET", url, headers=headers)
  response = connection.getresponse()
  # print the http status code returned. Expecting 200
  print "status: " + str(response.status), response.reason
  #print the response headers
  print response.getheaders()
  #print the response body
  print response.read()

#set endpoint and add your API key
url = "https://api.idxbroker.com/leads/lead"
headers = {"Content-Type":"application/x-www-form-urlencoded", "accesskey":"YourAPIKeyHere"}
#run function
main(url, headers)

File: GetLeads.php
<?php

// access URL and request method
$url = 'https://api.idxbroker.com/leads/lead';
$method = 'GET';

// headers (required and optional)
$headers = array(
	'Content-Type: application/x-www-form-urlencoded', // required
	'accesskey: abcdefghijklmnopqrstuvwx', // required - replace with your own
	'outputtype: json' // optional - overrides the preferences in our API control page
);

// set up cURL
$handle = curl_init();
curl_setopt($handle, CURLOPT_URL, $url);
curl_setopt($handle, CURLOPT_HTTPHEADER, $headers);
curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
curl_setopt($handle, CURLOPT_SSL_VERIFYHOST, false);
curl_setopt($handle, CURLOPT_SSL_VERIFYPEER, false);

if ($method != 'GET')
	curl_setopt($handle, CURLOPT_CUSTOMREQUEST, $method);

// exec the cURL request and returned information. Store the returned HTTP code in $code for later reference
$response = curl_exec($handle);
$code = curl_getinfo($handle, CURLINFO_HTTP_CODE);

if ($code >= 200 || $code < 300)
	$response = json_decode($response,true);
else
	$error = $code;


?>
